src/component/common/Splash/Splash.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button } from '@material-ui/core';
+import { useStyles } from './Splash.styles';
+import {
+    FiberManualRecord,
+    FiberManualRecordOutlined,
+    CloseOutlined,
+} from '@material-ui/icons';
+import { useState } from 'react';
+import ConditionallyRender from '../ConditionallyRender';
+
+interface ISplashProps {
+    components: React.ReactNode[];
+    onFinish: Function;
+}
+
+const Splash: React.FC<ISplashProps> = ({
+    components,
+    onFinish,
+}: ISplashProps) => {
+    const styles = useStyles();
+    const [counter, setCounter] = useState(0);
+
+    const onNext = () => {
+        if (counter === components.length - 1) {
+            onFinish(false);
+        }
+        setCounter(counter + 1);
```

Comment:
I think we should add a guard here for when to setCounter (don't increase it if we are already at the last page). Or return from the if clause above.

Reply:
Make sense.